builder/amazon/common/regions.go

```diff
@@ -1,29 +1,26 @@
 package common
 
-// aws ec2 describe-regions --query 'Regions[].{Name:RegionName}' --output text | sed 's/\(.*\)/"\1",/' | sort
+import (
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
 func listEC2Regions() []string {
-	return []string{
-		"ap-northeast-1",
-		"ap-northeast-2",
-		"ap-northeast-3",
-		"ap-south-1",
-		"ap-southeast-1",
-		"ap-southeast-2",
-		"ca-central-1",
-		"eu-central-1",
-		"eu-west-1",
-		"eu-west-2",
-		"eu-west-3",
-		"sa-east-1",
-		"us-east-1",
-		"us-east-2",
-		"us-west-1",
-		"us-west-2",
-		// not part of autogenerated list
-		"us-gov-west-1",
-		"cn-north-1",
-		"cn-northwest-1",
+	var regions []string
+	// append regions that are not part of autogenerated list
+	regions = append(regions, "us-gov-west-1", "cn-north-1", "cn-northwest-1")
```

Comment:
I think `ap-northeast-3` (Osaka) should be appended here as well - I don't see it in the list returned by `describe-regions` as it is a 'Local Region'

Reply:
If the credential in use has access to `ap-northeast-3` then it will be returned by `describe-regions`. You can request access to the region if you need it, https://aws.amazon.com/about-aws/global-infrastructure/#Asia_Pacific.

Reply:
Maybe we shouldn't append anything in that case, because according to credentials it will return the regions wdyt ?

Reply:
If I were a maintainer on this project, that's how I would do it. :) I don't think it is necessary to hardcode any regions at all.

Reply:
> If the credential in use has access to ap-northeast-3 then it will be returned by describe-regions

If that's definitely the case then dropping the hard coded regions is certainly the way to go

Reply:
You're probably going to have to hard code some regions somewhere/mock the results returned from `describe_regions` though in order to fix those failing tests...